packages/discovery-provider/src/tasks/entity_manager/entities/email.py

```diff
@@ -0,0 +1,260 @@
+from typing import List, Optional
+
+from src.exceptions import IndexingValidationError
+from src.models.users.email import EmailAccessKey, EmailEncryptionKey, EncryptedEmail
+from src.tasks.entity_manager.utils import (
+    Action,
+    EntityType,
+    ManageEntityParameters,
+    validate_signer,
+)
+from src.utils.structured_logger import StructuredLogger
+
+logger = StructuredLogger(__name__)
+
+
+def validate_email_metadata(params: ManageEntityParameters) -> None:
+    """
+    Validates the required fields in email metadata
+    Raises IndexingValidationError if validation fails
+    """
+    try:
+        if not isinstance(params.metadata, dict):
+            raise IndexingValidationError("Email metadata must be a dictionary")
+
+        primary_user_id = params.metadata.get("primary_user_id")
+        if not primary_user_id:
+            raise IndexingValidationError("primary_user_id is required")
+
+        if params.action in [Action.ADD_EMAIL, Action.UPDATE_EMAIL]:
+            required_fields = [
+                "primary_user_id",
+                "email_owner_user_id",
+                "encrypted_email",
+                "encrypted_key",
+            ]
+            missing_fields = [
+                field for field in required_fields if not params.metadata.get(field)
+            ]
+            if missing_fields:
+                raise IndexingValidationError(
+                    f"Missing required fields: {', '.join(missing_fields)}"
+                )
```

Comment:
else throw?

Reply:
not sure i understand

Reply:
sorry, if the action is not ADD_EMAIL or UPDATE_EMAIL, what should we do here?